src/Discovery/CollaboraDiscoveryFetcher.php

```diff
@@ -0,0 +1,196 @@
+<?php
+
+/*
+ * Copyright the Collabora Online contributors.
+ *
+ * SPDX-License-Identifier: MPL-2.0
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+declare(strict_types=1);
+
+namespace Drupal\collabora_online\Discovery;
+
+use Drupal\collabora_online\Exception\CollaboraNotAvailableException;
+use Drupal\Component\Datetime\TimeInterface;
+use Drupal\Core\Cache\CacheBackendInterface;
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\Config\ImmutableConfig;
+use GuzzleHttp\ClientInterface;
+use GuzzleHttp\RequestOptions;
+use Psr\Http\Client\ClientExceptionInterface;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\DependencyInjection\Attribute\Autowire;
+use Symfony\Component\ErrorHandler\ErrorHandler;
+
+/**
+ * Creates a discovery value object.
+ */
+class CollaboraDiscoveryFetcher implements CollaboraDiscoveryFetcherInterface {
+
+  public const DEFAULT_CID = 'collabora_online.discovery';
+
+  public function __construct(
+    #[Autowire(service: 'logger.channel.collabora_online')]
+    protected readonly LoggerInterface $logger,
+    protected readonly ConfigFactoryInterface $configFactory,
+    protected readonly ClientInterface $httpClient,
+    #[Autowire(service: 'cache.default')]
+    protected readonly CacheBackendInterface $cache,
+    #[Autowire(value: self::DEFAULT_CID)]
```

Comment:
There's no need to have this customisable when instantiating the service.

Reply:
Having a dynamic parameter allows to use this class for a different instance of Collabora, with its own cache cid.
It is not needed for now though, we can add it when we need to.

Reply:
ok, i checked core: most of the times when a cid is dynamic, it has one hard-coded part and one dynamic part based on e.g. current theme or some other dynamic value.

So no need to make it settable in this version.

Reply:
done